Sources/Sentry/SentryHttpTransport.m

```diff
@@ -259,6 +260,11 @@ - (NSDate *)parseRetryAfterHeader:(NSString * __nullable)retryAfterHeader {
     // parsing as double/seconds failed, try to parse as date
     NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
```

Comment:
I know this wasn't introduced in this PR and but before we go GA
I don't think we need to create a dateformatter for every call in `parseRetryAfterHeader`.
Initializing one is expensive and maybe we should only do it once per transport.

Reply:
> NSDateFormatter is not thread safe

https://developer.apple.com/documentation/foundation/nsdateformatter

Sounds like we either need to create one per request or we'd need to create a pool

Reply:
Also worth noting that *today* Sentry returns seconds, not date.

Reply:
I read it again now and I skimmed over important bits:

> In macOS 10.9 and later NSDateFormatter is thread safe so long as you are using the modern behavior in a 64-bit app.
On earlier versions of the operating system, or when using the legacy formatter behavior or running in 32-bit in macOS, NSDateFormatter is not thread safe

Reply:
I am going to address this issue in another PR. I am already on it.

Reply:
I opened another PR for this https://github.com/getsentry/sentry-cocoa/pull/440